swupd/create_manifests.go

```diff
@@ -258,6 +254,14 @@ func CreateManifests(version uint32, minVersion bool, format uint, statedir stri
 
 	// write manifests then add them to the MoM
 	for _, bMan := range newManifests {
+		// full is just another manifest, grab it from here and maximize versions
+		if bMan.Name == "full" {
+			maximizeFull(bMan, newManifests)
```

Comment:
I think this maximize step should happen before this loop, but I might be missing something.

The potential problem I'm seeing is newManifests contains "a", "full" and "b".

- "a" gets sorted by version then name
- "full" causes resort of "a" and "b" to be name
- "full" itself is sorted by version then name
- "b" is sorted by version then name

At the end of the loop, "a" sorting is incorrect.